components/authz-service/storage/postgres/migration/sql/59_use_db_id_as_primary_key_in_iam_projects.up.sql

```diff
@@ -0,0 +1,26 @@
+BEGIN;
+
+ALTER TABLE iam_projects
+    DROP CONSTRAINT iam_projects_pkey; -- 'id' primary key
+ALTER TABLE iam_projects
+    ADD PRIMARY KEY (db_id); -- implies unique
```

Comment:
TIL: This is a [unique table constraint](https://www.postgresql.org/docs/8.1/ddl-constraints.html) (see section 5.3.3.) vs a table constraint (when you use the `UNIQUE` keyword).

Interestingly, when doing more research on table vs column unique constraint -- looks like the difference doesn't actually matter to postgres?

> We say that the first two constraints are column constraints, whereas the third one is a table constraint because it is written separately from any one column definition. Column constraints can also be written as table constraints, while the reverse is not necessarily possible, since a column constraint is supposed to refer to only the column it is attached to. (PostgreSQL doesn't enforce that rule, but you should follow it if you want your table definitions to work with other database systems.)

https://stackoverflow.com/questions/7455839/postgresql-difference-between-table-constraint-and-column-constraint

Reply:
But, column constraints and table constraints are not actually different. When dealing with a _single_ column they are just two different ways to write exactly the same thing. When trying to apply uniqueness across _multiple_ columns, you must write it as a table constraint, but really only because of syntactical limitations.